packages/medusa/src/loaders/api.ts

```diff
@@ -28,6 +28,28 @@ export default async ({ app, container, plugins }: Options) => {
     ContainerRegistrationKeys.CONFIG_MODULE
   )
 
+  /**
+   * Always load plugin routes before the Medusa core routes, since it
+   * will allow the plugin to define routes with higher priority
+   * than Medusa. Here are couple of examples.
+   *
+   * - Plugin registers a route called "/products/active"
+   * - Medusa registers a route called "/products/:id"
+   *
+   * Now, if Medusa routes gets registered first, then the "/products/active"
+   * route will never be resolved, because it will be handled by the
+   * "/products/:id" route.
+   */
+  await Promise.all(
+    plugins.map(async (pluginDetails) => {
+      return new RoutesLoader({
+        app: app,
+        configModule,
+        rootDir: path.join(pluginDetails.resolve, "api"),
+      }).load()
+    })
+  )
```

Comment:
sugg: to wrap this baby up with a user friendly error

Reply:
Agree 100%.

But usually we have these sorts of loaders in many places and most of them are not wrapped in `try/catch` and then showing a user friendly error.

Maybe, we have to find all of them and settle on a pattern to use.